### PR TITLE
[JBIDE-22592] always fire change for same properties list instance

### PR DESCRIPTION
**Pull request by adietish**

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/fromtemplate/ApplicationSourceFromTemplateModel.java [hunk 1]
```diff
@@ -181,54 +180,56 @@ public boolean isParameterModified(IParameter param) {
...
 		Map<String, String> projectParams = getProjectParameters(project);
 
-		List<IParameter> newParameters = originalParameters.stream().map(p -> { 
-			IParameter clone = p.clone();
-			String value = projectParams.get(clone.getName());
+		originalParameters.forEach(p -> { 
+			String value = projectParams.get(p.getName());
 			if (value != null) {
-				clone.setValue(value);
+				p.setValue(value);
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/fromtemplate/ApplicationSourceFromTemplateModel.java]: DONT clone the parameters, set the value to the existing parameter.

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/fromtemplate/ApplicationSourceFromTemplateModel.java [hunk 2]
```diff
@@ -181,54 +180,56 @@ public boolean isParameterModified(IParameter param) {
...
+	public void setParameters(Collection<IParameter> parameters) {
+		List<IParameter> oldParameters = new ArrayList<>(this.parameters);
+		if (parameters == null) {
+			this.parameters.clear();
+		} else {
+			this.originalValueMap = toMap(parameters);
+			List<IParameter> newParameters = new ArrayList<>();
+			newParameters.addAll(injectProjectParameters(eclipseProject, parameters));
+			this.parameters.clear();
+			this.parameters.addAll(newParameters);
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/fromtemplate/ApplicationSourceFromTemplateModel.java]: add the parameters to the existing list, **DONT** replace the list

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/fromtemplate/ApplicationSourceFromTemplateModel.java [hunk 3]
```diff
@@ -141,8 +140,11 @@ private void handleSelectedAppSource(PropertyChangeEvent evt) {
 		if(evt.getNewValue() instanceof IApplicationSource
 				&& ResourceKind.TEMPLATE.equals(((IApplicationSource) evt.getNewValue()).getKind())){
 			IApplicationSource source = (IApplicationSource) evt.getNewValue();
-			this.template = (ITemplate) source.getSource();
-			updateTemplateParameters(template);
+			ITemplate newTemplate = (ITemplate) source.getSource();
+			if (!Objects.equals(newTemplate, this.template)) {
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/fromtemplate/ApplicationSourceFromTemplateModel.java]: dont update template parameters if we already have the very same template already set

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/fromtemplate/ApplicationSourceFromTemplateModel.java [hunk 4]
```diff
@@ -181,54 +184,60 @@ public boolean isParameterModified(IParameter param) {
...
+			List<IParameter> newParameters = new ArrayList<>();
+			newParameters.addAll(injectProjectParameters(eclipseProject, parameters));
+			this.parameters.clear();
+			this.parameters.addAll(newParameters);
+			Collections.sort(this.parameters, new TemplateParameterViewerUtils.ParameterNameComparator());
+		}
+		firePropertyChange(PROPERTY_PARAMETERS, oldParameters, this.parameters);
+
+		// update selected parameter
+		setSelectedParameter(getSelectedParameterOrDefault());
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/fromtemplate/ApplicationSourceFromTemplateModel.java]: update selected parameter after new parameters have been set.

**Comment by fbricon** [comment]
+1

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/fromtemplate/ApplicationSourceFromTemplateModel.java [hunk 5]
```diff
@@ -181,54 +184,60 @@ public boolean isParameterModified(IParameter param) {
 	}
 
 	@Override
-	public void setParameters(List<IParameter> parameters) {
+	public void setParameters(Collection<IParameter> parameters) {
+		List<IParameter> oldParameters = new ArrayList<>(this.parameters);
+		if (parameters == null) {
+			this.parameters.clear();
```
Comment (scabanovich) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/fromtemplate/ApplicationSourceFromTemplateModel.java]: May this.originalValueMap be cleared here, too?

**Pull request merged**